### PR TITLE
Add Basic CINN Runner Class

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

New features
### PR changes

APIs
### Describe

Add basic CINN entrance class.

There are 3 classes in this PR: `CinnRunner`, `CinnCacheKey`, `CinnCompiledObject`

`CinnRunner` is the main tool class which stores a cache from `CinnCacheKey` to `CinnCompiledObject` .
`CinnCacheKey` use `Graph`, input names, input shapes as cache key.
`CinnCompiledObject` will be the communication between Paddle and CINN, we will add more functions here after we have CINN interface.

### CODE REVIEW

File: paddle/fluid/framework/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ add_subdirectory(details)
 add_subdirectory(fleet)
 add_subdirectory(io)
 add_subdirectory(new_executor)
+add_subdirectory(cinn)
```
Comment (CtfGo) [inline paddle/fluid/framework/CMakeLists.txt]: I think the name of this subdirectory is not clear enough to indicate we are the user of cinn and link it, someone like` pd2cinn` or `compiler_adaptor` maybe better.

**Review by CtfGo — COMMENTED** [review]
In General, this pr LGTM, though it may need a little update afterwards to match cinn interface.

File: paddle/fluid/framework/CMakeLists.txt (hunk 1 above)
Comment (zhhsplendid) [reply paddle/fluid/framework/CMakeLists.txt]: Agree, but rename to `paddle2cinn` to match naming style of other modules.

**Comment by zhhsplendid** [comment]
> In General, this pr LGTM, though it may need a little update afterwards to match cinn interface.

Yes, I agree

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**